electrolyzer/lcoh.py

```diff
@@ -0,0 +1,431 @@
+            Feedstock:
+                functions used:
+                    -calc_feedstock_costs
+                -user_inputs:
+                    water_feedstock_cost
+                    water_per_kgH2
+                    dt
+                -external_data:
+                    lcoe
+                -simulation_data:
+                    'kg_rate' -> to calculate H20
+                    'power_signal'
+                    'curatilment'
+                -simulation data scaled to represent a year
+                -Water feedstock (requires H2 production)
+                -Electricity Feedstock (requires LCOE)
+            Stack Replacements:
+                functions used:
+                    -calc_stack_replacement_costs (main)
+                    -calc_stack_replacement_schedule (called by main)
+                -user_inputs:
+                    plant_life
+                    d_eol
+                    n_stacks
+                    stack_rating_kW
+                    stack_replacement_percent
+                -simulation_data:
+                    deg_state
+                    sim_length (time)
+                -first determine stack replacement schedule
+                -cost for 1 stack replacement is total uninstalled CapEx
+                    divided by number of stacks
+                -multiply cost by replacement schedule
+                -OUPUTS: stack replacement cost per year (array of length plant_life)
+        CapEx Costs:
+            functions used:
+                -pem_capex_calc
+                -bop_capex_calc
+            -split into BOP and PEM
+                -user_inputs:
+                    capex_learning_rate
+                    ref_cost_bop & ref_cost_pem
+                    ref_size_bop & ref_size_pem
+                    stack_rating_kW
+                    n_stacks
+                    install_factor
+
+        H2 Production:
+            -done in LCOH calculation
+            -simulation data scaled to represent a year
+
+
+
+
+        """
+        self.CapEx_summary = {}
+        self.StackReplacement_summary = {}
+        self.StackReplacement_schedule = {}
+        self.Feedstock_summary = {}
+        self.OpEx_summary = {}
+        self.LCOH_summary = {}
+        self.plant_life_yrs = self.plant_params["plant_life"]
+        if self.plant_params["pem_location"] == "onshore":
+            self.os = 0
+        else:
+            self.os = 1
+
+        self.IF = self.finances["install_factor"]
+        self.DR = self.finances["discount_rate"]
+        self.sim_length_hrs = 0
+        self.plant_rating_kW = 0
+        self.n_stacks = 0
+        self.stack_rating_kW = 0
+
+        self.d_sim = np.zeros(7)
+        self.power_kW_avail = np.zeros(7)
+        self.power_kW_curtailed = np.zeros(7)
+        self.kg_produced = np.zeros(7)
+
+        self.electrical_feedstock_cost = 0.0
+        self.dt = 0
+
+    def get_simulation_info(self, elec_sys, elec_df, lcoe):
```

Comment:
I see why you did this, since not all results are contained in the `Supervisor` instance after the run. Luckily I think we can refactor this so that we can do all of this on initialization, without depending on passing potentially large DataFrame objects. This will help us make `optimize=True` possible without too much extra hassle